### ISSUE TITLE
[wg/gpu] GPU for the Web WG recharter

### ISSUE THREAD

**Issue by tidoust** [w3c/strategy#480]
New charter proposal, reviewers please take note.

# Charter Review

[Charter](https://gpuweb.github.io/admin/wg-charter.html)

[diff from charter template](https://services.w3.org/htmldiff?doc1=https://w3c.github.io/charter-drafts/charter-template.html&doc2=https://gpuweb.github.io/admin/wg-charter.html)

If applicable:

[diff from previous charter](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2F2022%2F11%2Fgpuweb-charter.html&doc2=https%3A%2F%2Fgpuweb.github.io%2Fadmin%2Fwg-charter.html)

[chair dashboard](https://www.w3.org/PM/Groups/chairboard.html?gid=wg/gpu)

What kind of charter is this? Check the relevant box / remove irrelevant branches. 

* New
 - [ ] New WG 
 - [ ] New IG
 - If this is a new WG or IG charter request, link to [Advance Notice](), and any issue discussion: 

* Existing 
 - [x] Existing WG recharter 
 - [ ] Existing IG recharter
 - If this is a charter extension or revision, any issue discussion:

Horizontal Reviews: apply the Github label "Horizontal review requested" to request reviews for accessibility (a11y), internationalization (i18n), privacy, security, and TAG. **Also add a "card" for this issue to the [Strategy Funnel](https://github.com/orgs/w3c/projects/97/views/2).**

## Communities suggested for outreach

Groups identified in the Coordination section.

## Known or potential areas of concern

None. Proposed scope and deliverables are the same. Main proposed change is the switch to perpetual CR mode. That said, note that the GPU for the Web WG is still reviewing the draft and may adjust it. Internal review in the group should conclude at the upcoming group's F2F end of October.

Where would charter proponents like to see issues raised? In the [gpuweb/admin](https://github.com/gpuweb/admin) repository (or this strategy funnel issue)

## Anything else we should think about as we review?

The group should soon request transition of the WebGPU and WGSL specifications to Candidate Recommendation.

Cc: @Kangz @kdashg

**Comment by simoneonofri** [comment]
From a Security point of view, the considerations on WebGPU seem to me to be well written.

I was pondering, reading [this paper](https://arxiv.org/pdf/2409.01824) and also considering the structure of browser processes, whether we can add some mitigation in this regard.

**Comment by himorin** [comment]
no comment or request from i18n

**Comment by ruoxiran** [comment]
no comment or request from APA.

**Comment by plehegar** [comment]
Privacy is wondering about the status of https://github.com/gpuweb/gpuweb/issues/3101

**Comment by tidoust** [comment]
> I was pondering, reading [this paper](https://arxiv.org/pdf/2409.01824) and also considering the structure of browser processes, whether we can add some mitigation in this regard.

@simoneonofri Does that warrant charter updates though? The charter already states in its Scope section that the group "will investigate and document threat mitigation strategies for the API, notably to address fingerprinting issues and to prevent unauthorized use of computational resources". If additional mitigations are warranted, the group would be happy to consider them. Linking to the document you point out from the charter seems overly specific.

**Comment by tidoust** [comment]
> Privacy is wondering about the status of [gpuweb/gpuweb#3101](https://github.com/gpuweb/gpuweb/issues/3101)

That's one of the few remaining open issues that the group is aware of and will provide an answer for before requesting transition to CR. The group does not anticipate going beyond what was already put in place in response to the issue (restriction to 5 bits revealed, correlated with information that can be obtained through other means).

**Comment by plehegar** [comment]
PING is fine moving forward with the charter.

**Comment by tidoust** [comment]
Following discussions within the group and with @ianbjacobs and @dontcallmedom, the draft charter was further adjusted to reflect on the actual work mode and get back to having the WG drive the technical work on the specs (the text previously said that the CG was driving the technical work). See related update: https://github.com/gpuweb/admin/pull/24

The change essentially gets us back to regular boilerplate text found in other working group charters. I'm going to assume that it does not affect horizontal reviews and proceed to the next chartering step. Feel free to chime in if you feel otherwise!

**Comment by himorin** [comment]
I suppose 'End date' in the header table should be replaced with something like `start date + 2 years`, but not `2024-11-30`.

**Comment by svgeesus** [comment]
> I see no "Motivation and Background" section, could it be added?

True, although at least all the text in the Scope section is in fact scope. The Motivation section was originally added to stop people stuffing Scope with introductory prose, which is not the case here.

**Comment by tidoust** [comment]
> The Motivation section was originally added to stop people stuffing Scope with introductory prose

I thought so as well. I guess the "Motivation and Background" section in the charter template should have some "motivation and background" text to explain why it's there ;)

Anyway, I prepared a pull request to add a short section, see https://github.com/gpuweb/admin/pull/26. The GPU for the Web CG maintains an [explainer](https://gpuweb.github.io/gpuweb/explainer/), linked from that suggested text, that goes deeper into details.

**Comment by tidoust** [comment]
[AC Review started](https://lists.w3.org/Archives/Member/w3c-ac-members/2024OctDec/0050.html) (member-only link)

**Comment by plehegar** [comment]
[VOTE by 2024-12-20/21: Proposed Charter for the GPU for the Web Working Group; Current Charter Extended](https://lists.w3.org/Archives/Member/w3c-ac-members/2024OctDec/0050.html)

**Comment by plehegar** [comment]
[Announced](https://lists.w3.org/Archives/Member/w3c-ac-members/2025JanMar/0019.html)
